docs(terraform): improve documentation for filtering by inline comments (#6284)
aquasecurity/trivy: docs/docs/scanner/misconfiguration/index.md

### Skipping resources by inline comments

Trivy supports ignoring misconfigured resources by inline comments for Terraform configuration files only.

> In cases where Trivy can detect comments of a specific format immediately adjacent to resource definitions, it is possible to filter/ignore findings from a single point of resource definition (in contrast to `.trivyignore`, which has a directory-wide scope on all of the files scanned). The format for these comments is `trivy:ignore:<rule>` or `tfsec:ignore:<rule>` immediately following the format-specific line-comment [token](https://developer.hashicorp.com/terraform/language/syntax/configuration#comments).

Reviewer: ```suggestion
In cases where Trivy can detect comments of a specific format immediately adjacent to resource definitions, it is possible to ignore findings from a single source of resource definition (in contrast to `.trivyignore`, which has a directory-wide scope on all of the files scanned). The format for these comments is `trivy:ignore:<rule>` immediately following the format-specific line-comment [token](https://developer.hashicorp.com/terraform/language/syntax/configuration#comments).
```

Author: Done https://github.com/aquasecurity/trivy/pull/6284/commits/857f3349fecabb6b7f00ab38646e69c677149953

> The ignore rule must contain one of the possible check IDs that can be found in its metadata: ID, short code or alias. The `id` from metadata is not case sensitive, you can specify `AVD-AWS-0089` or `avd-aws-0089` and it will work the same way.

Reviewer: ```suggestion
The ignore rule must contain one of the possible check IDs that can be found in its metadata: ID, short code or alias. The `id` from metadata is not case sensitive, you can specify `AVD-AWS-0089` or `avd-aws-0089` as it's case insensitive.
```

Author: Done https://github.com/aquasecurity/trivy/pull/6284/commits/857f3349fecabb6b7f00ab38646e69c677149953

For example, to ignore a misconfiguration ID `AVD-GCP-0051` in a Terraform HCL file:

```terraform
#trivy:ignore:AVD-GCP-0051
resource "google_container_cluster" "example" {
  name     = var.cluster_name
  location = var.region
}
```

You can add multiple ignores on the same comment line:
```terraform
#trivy:ignore:AVD-GCP-0051 trivy:ignore:AVD-GCP-0053
resource "google_container_cluster" "example" {
  name     = var.cluster_name
  location = var.region
}
```

You can also specify a long ID, which is formed as follows: `<provider>-<service>-<short-code>`.

As an example, consider the following check metadata:

```yaml
# custom:
  # id: AVD-AWS-0089
  # avd_id: AVD-AWS-0089
  # provider: aws
  # service: s3
  # severity: LOW
  # short_code: enable-logging
```

Long ID would look like the following: `aws-s3-enable-logging`.

> !!! tip
>     Long ID is preferred because it is a readable format.

Reviewer: While it's more readable, I would argue that it's more error prone so I'm not sure if we should suggest users to use it over something like AVD ID.
```suggestion
```

Author: Done https://github.com/aquasecurity/trivy/pull/6284/commits/857f3349fecabb6b7f00ab38646e69c677149953
